### PR TITLE
Extract hard links from tar

### PR DESCRIPTION
**Pull request by priyawadhwa**

### CODE REVIEW

File: pkg/util/tar_utils.go [hunk 1]
```diff
@@ -119,6 +122,31 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
...
+	for target, linkname := range hardlinks {
+		logrus.Info("Resolving hard links.")
+		if _, err := os.Stat(linkname); !os.IsNotExist(err) {
+			// If it exists, create the hard link
+			if err := os.Link(linkname, target); err != nil {
+				logrus.Warnf("Unable to create hard link from %s to %s: %v", linkname, target, err)
+			}
+			logrus.Debugf("Created hard link from %s to %s", linkname, target)
+		} else {
+			logrus.Warnf("Unable to create hard link from %s to %s", linkname, target)
```
[the remark this answers is not in the document]
Comment (priyawadhwa) [reply pkg/util/tar_utils.go]: I can't think of any case in which it should fail at this point, so an error would probably make more sense here

File: pkg/util/tar_utils.go [hunk 2]
```diff
@@ -119,6 +122,31 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
 			if err = os.Symlink(header.Linkname, target); err != nil {
 				logrus.Errorf("Failed to create symlink between %s and %s: %s", header.Linkname, target, err)
 			}
+		case tar.TypeLink:
+			linkname := filepath.Join(path, header.Linkname)
+			// Check if the linkname already exists
+			if _, err := os.Stat(linkname); !os.IsNotExist(err) {
+				// If it exists, create the hard link
+				if err := os.Link(linkname, target); err != nil {
+					logrus.Warnf("Failed to create hard link between %s and %s: %v", linkname, target, err)
```
[the remark this answers is not in the document]
Comment (priyawadhwa) [reply pkg/util/tar_utils.go]: done

File: pkg/util/tar_utils.go [hunk 3]
```diff
@@ -119,11 +122,38 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
...
+		}
+	}
+
+	for target, linkname := range hardlinks {
+		logrus.Info("Resolving hard links.")
+		if _, err := os.Stat(linkname); !os.IsNotExist(err) {
+			// If it exists, create the hard link
+			resolveHardlink(linkname, target)
+		} else {
+			logrus.Errorf("Unable to create hard link from %s to %s", linkname, target)
```
Comment (nkubala) [inline pkg/util/tar_utils.go]: actually, can you return this error? we should be sending these back to the top level and handling them there

File: pkg/util/tar_utils.go [hunk 4]
```diff
@@ -119,11 +122,38 @@ func unpackTar(tr *tar.Reader, path string, whitelist []string) error {
...
 		}
 	}
 	return nil
 }
 
+func resolveHardlink(linkname, target string) {
```
Comment (nkubala) [inline pkg/util/tar_utils.go]: can you return an error from this function? then you can just have the error message here and get rid of the one above. you can use `errors.Wrap(err, errMsg)`

**Pull request merged**